command/format/diff.go

```diff
@@ -742,20 +740,34 @@ func (p *blockBodyDiffPrinter) writeValueDiff(old, new cty.Value, indent int, pa
 			p.buf.WriteString(strings.Repeat(" ", indent))
 			p.buf.WriteString("]")
 			return
-
 		case ty.IsListType() || ty.IsTupleType():
 			p.buf.WriteString("[")
 			if p.pathForcesNewResource(path) {
 				p.buf.WriteString(p.color.Color(forcesNewResourceCaption))
 			}
 			p.buf.WriteString("\n")
 
-			elemDiffs := ctySequenceDiff(old.AsValueSlice(), new.AsValueSlice())
-			for _, elemDiff := range elemDiffs {
-				p.buf.WriteString(strings.Repeat(" ", indent+2))
-				p.writeActionSymbol(elemDiff.Action)
-				p.writeValue(elemDiff.Value, elemDiff.Action, indent+4)
-				p.buf.WriteString(",\n")
+			if ty.IsTupleType() && ty.Length() > 0 && ty.TupleElementType(0).IsObjectType() {
```

Comment:
Here I just assumed that we'll never have tuples of multiple types, but I am not sure if it's a safe assumption at all. 🤔

Reply:
Indeed it is _not_; the primary purpose of tuples is to allow separate types per element, and it's particularly relevant in a JSON context because cty tuple is the representation of a JSON array, and so users can write `["a", {}]` in a JSON argument which would become a `tuple([string, object({})]` value once parsed.

To get correct behavior here we may need to somehow combine the object-sequence behavior into `ctySequenceDiff`, and have it be the one to deal with any awkward single-element type changes.